docs/data-sources/gateway.md

```diff
@@ -0,0 +1,38 @@
+---
+page_title: "Equinix Metal: Metal Gateway"
+subcategory: ""
+description: |-
+  Retrieve Equinix Metal Gateways
+---
+
+# metal\_gateway
+
+Use this datasource to retrieve Metal Gateway resources in Equinix Metal.
+
+## Example Usage
+
+```hcl
+# Create Metal Gateway for a VLAN with a private IPv4 block with 8 IP addresses
+
+resource "metal_vlan" "test" {
+  description = "test VLAN in SV"
+  metro       = "sv"
+  project_id  = local.project_id
+}
+
+data "metal_gateway" "test" {
```

Comment:
We may want to offer a gateway_id way to lookup metal_reserved_ip_block datasources. 
https://registry.terraform.io/providers/equinix/metal/latest/docs/data-sources/reserved_ip_block

I'll open a new issue for that.

Reply:
It might also be helpful to lookup a metal_gateway by IP block or reservation. I'll open an issue for that.